### PR TITLE
Add feature flag package

### PR DESCRIPTION
**Pull request by teimurjan**
## What's done

Added a new feature flag package to aggregate all the apps' feature flags.

## How to test

Run `yarn test`.

## Tasks

- [x] Have you written tests for new code (if applicable)?
- [x] Have you tested the changes on all the platforms?
  - [x] iOS
  - [x] Android
  - [x] Web
- [x] Have you added stories to demonstrate the new functionality (if there is any)?

## Demo (screenshots, recordings, etc.)

[Check **Usage** & **Advanced Usage** in the README](https://github.com/TuringAdvisoryGroup/tryrolljs/blob/7d796ba11a354c1e44f3f06389cd7c61d947e838/packages/feature-flag/README.md)

### CODE REVIEW

File: packages/feature-flag/src/index.test.tsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+import { PropsWithChildren } from 'react'
+import { FeatureFlag } from './types'
+import { useFeatureFlag, FeatureFlagProvider } from './index'
+
+const getWrapper =
+  (flags: FeatureFlag[]) =>
+  ({ children }: PropsWithChildren<{}>) =>
+    <FeatureFlagProvider flags={flags}>{children}</FeatureFlagProvider>
+
+describe('useFeatureFlag', () => {
```
Comment (teimurjan) [inline packages/feature-flag/src/index.test.tsx:11]: Any unseen test cases are welcome 🙂

File: packages/feature-flag/src/index.tsx [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    dispatch({
+      type: 'complete',
+      payload: featureFlags,
+    })
+  }
+
+  useEffect(() => {
+    initialize()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
```
Comment (teimurjan) [inline packages/feature-flag/src/index.tsx:132]: Evaluating feature flags once. I think it's better than dealing with memorization of the flags & all the functions.

File: packages/feature-flag/src/types.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  defaultValue: FeatureFlagMap
+  value: AsyncGroupFeatureFlagValue
+}
+export type ComputedFeatureFlag = {
+  type: 'computed'
+  name: string
+  value: ComputedFeatureFlagValue
+}
+
+export type FeatureFlag =
```
Comment (teimurjan) [inline packages/feature-flag/src/types.ts:31]: We'll be able to extend it later.

**Review by diegonzs-roll — APPROVED** [review]
LGTM! 👍🏻

**Pull request merged**